core/local/local_test.go

```diff
@@ -96,6 +101,479 @@ func TestExecutionSchedulerRun(t *testing.T) {
+	defer tb.Cleanup()
+	sr := tb.Replacer.Replace
+
+	script := sr(`
+	import http from "k6/http";
+	import ws from 'k6/ws';
+	import { Counter } from 'k6/metrics';
+	import { check, group } from 'k6';
+
+	let errors = new Counter('errors');
+
+	export let options = {
+		execution: {
+			scenario1: {
+				type: 'per-vu-iterations',
+				vus: 1,
+				iterations: 1,
+				gracefulStop: '0.5s',
+				exec: 's1func',
+				env: { TESTVAR1: 'scenario1' },
+				tags: { testtag1: 'scenario1' },
+			},
+			scenario2: {
+				type: 'shared-iterations',
+				vus: 1,
+				iterations: 1,
+				gracefulStop: '0.5s',
+				exec: 's2func',
+				env: { TESTVAR2: 'scenario2' },
+				tags: { testtag2: 'scenario2' },
+			},
+			scenario3: {
+				type: 'per-vu-iterations',
+				vus: 1,
+				iterations: 1,
+				gracefulStop: '0.5s',
+				exec: 's3funcWS',
+				env: { TESTVAR3: 'scenario3' },
+				tags: { testtag3: 'scenario3' },
+			},
+		}
+	}
+
+	function checkVar(name, expected) {
+		if (__ENV[name] !== expected) {
+		    console.error('Wrong ' + name + " env var value. Expected: '"
+						+ expected + "', actual: '" + __ENV[name] + "'");
+			errors.add(1);
+		}
+	}
+
+	export function s1func() {
+		checkVar('TESTVAR1', 'scenario1');
+		checkVar('TESTVAR2', undefined);
+		checkVar('TESTVAR3', undefined);
+		checkVar('TESTGLOBALVAR', 'global');
+
+		http.get('HTTPBIN_IP_URL/', { tags: { reqtag: 'scenario1' }});
+	}
+
+	export function s2func() {
+		checkVar('TESTVAR1', undefined);
+		checkVar('TESTVAR2', 'scenario2');
+		checkVar('TESTVAR3', undefined);
+		checkVar('TESTGLOBALVAR', 'global');
+
+		http.get('HTTPBIN_IP_URL/', { tags: { reqtag: 'scenario2' }});
+	}
+
+	export function s3funcWS() {
+		checkVar('TESTVAR1', undefined);
+		checkVar('TESTVAR2', undefined);
+		checkVar('TESTVAR3', 'scenario3');
+		checkVar('TESTGLOBALVAR', 'global');
+
+		const customTags = { wstag: 'scenario3' };
+		group('wsgroup', function() {
+			const response = ws.connect('WSBIN_URL/ws-echo', { tags: customTags },
+				function (socket) {
+					socket.on('open', function() {
+						socket.send('hello');
+					});
+					socket.on('message', function(msg) {
+						if (msg != 'hello') {
+						    console.error("Expected to receive 'hello' but got '" + msg + "' instead!");
+							errors.add(1);
+						}
+						socket.close()
+					});
+					socket.on('error', function (e) {
+						console.log('ws error: ' + e.error());
+						errors.add(1);
+					});
+				}
+			);
+			check(response, { 'status is 101': (r) => r && r.status === 101 }, customTags);
+		});
+	}
+`)
+
+	runner, err := js.New(&loader.SourceData{
+		URL:  &url.URL{Path: "/script.js"},
+		Data: []byte(script)},
+		nil, lib.RuntimeOptions{Env: map[string]string{"TESTGLOBALVAR": "global"}})
+	require.NoError(t, err)
+
+	// Required for WS tests
+	opts := runner.GetOptions()
+	opts.Hosts = map[string]net.IP{
+		"httpbin.local": net.ParseIP("127.0.0.1"),
+	}
+	require.NoError(t, runner.SetOptions(opts))
```

Comment:
nit: Does this not work if it's just in the text options above ?

Reply:
Indeed :facepalm: Fixed in 8c5d8df3.